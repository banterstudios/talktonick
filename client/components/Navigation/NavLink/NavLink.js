import React, { Component } from 'react'
import glamorous from 'glamorous'
import Link from 'client/components/Gui/Link'
import { withRouter } from 'react-router-dom'
import { omit } from 'client/utils/objectUtils'

const NavLinkWrapper = glamorous.div(() => ({
  position: 'relative',
  width: '100%'
}))

const disAllowedProps = ['match', 'location', 'history', 'isActive']

@withRouter
export default class NavLink extends Component {
  checkIfActive = ({ location: { pathname = '' }, href }) => {
    return (pathname && (pathname === href))
  }

  render () {
    const {
      className,
      href,
      location,
      ...linkProps
    } = this.props

    const isActive = this.checkIfActive({ location, href })

    return (
      <NavLinkWrapper className={className}>
        <Link {...omit(linkProps, disAllowedProps)} href={href} isActive={isActive} />
      </NavLinkWrapper>
    )
  }
}
