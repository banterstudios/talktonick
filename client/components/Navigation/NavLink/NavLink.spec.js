import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NavLink from './NavLink'
import { MemoryRouter as Router } from 'react-router-dom'

describe('components/Navigation/NavLink', () => {
  describe('@renders', () => {
    it('should render correctly', () => {
      const component = shallow(<Router><NavLink /></Router>)

      expect(toJson(component.dive())).toMatchSnapshot()
      component.unmount()
    })
  })
})
