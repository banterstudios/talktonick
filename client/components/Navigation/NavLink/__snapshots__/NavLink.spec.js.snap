// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navigation/NavLink @renders should render correctly 1`] = `<withRouter(NavLink) />`;
